File: src/leetcode/python/LongestConsecutiveSequence_128.py
class LongestConsecutiveSequence_128(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        num = set(nums)
        res = 0
        for n in num:
            if n - 1 not in num:
                m = n + 1
                while m in num:
                    m += 1
                res = max(res, m - n)
        return res

File: src/leetcode/java/LongestConsecutiveSequence_128.java
package leetcode.java;

import java.util.HashMap;
 
public class LongestConsecutiveSequence_128 {
  public int longestConsecutive(int[] nums) {
    HashMap<Integer, Integer> map = new HashMap<>();
    int res = 0;
    for (int i : nums) {
      if (!map.containsKey(i)) {
        int left = map.containsKey(i - 1) ? map.get(i - 1) : 0;
        int right = map.containsKey(i + 1) ? map.get(i + 1) : 0;
        int sum = left + right + 1;
        map.put(i, sum);
        res = Math.max(res, sum);
        map.put(i - left, sum);
        map.put(i + right, sum);
      }
    }
    return res;
  }
}
